### ISSUE TITLE
systemctl set-property multiple values of IPAddressAllow= does not work while worked in 249.

### ISSUE THREAD

**Issue by gitzdnex** [systemd/systemd#34773]
### systemd version the issue has been seen with

256.6-1

### Used distribution

Archlinux

### Linux kernel version used

Linux 6.8.0-45-generic

### CPU architectures issue was seen on

x86_64

### Component

systemctl

### Expected behaviour you didn't see

On ubuntu 22.04 with systemd 249 i could run
```
$systemctl set-property systemd-resolved.service IPAddressDeny= IPAddressAllow= IPAddressDeny=any IPAddressAllow=2.2.2.2 IPAddressAllow=localhost
```
And then I would get
```
# /etc/systemd/system.control/systemd-resolved.service.d/50-IPAddressAllow.conf
# This is a drop-in unit file extension, created via "systemctl set-property"
# or an equivalent operation. Do not edit.
[Service]
IPAddressAllow=
IPAddressAllow=::1/128
IPAddressAllow=127.0.0.0/8
IPAddressAllow=2.2.2.2/32


# /etc/systemd/system.control/systemd-resolved.service.d/50-IPAddressDeny.conf
# This is a drop-in unit file extension, created via "systemctl set-property"
# or an equivalent operation. Do not edit.
[Service]
IPAddressDeny=
IPAddressDeny=::/0
IPAddressDeny=0.0.0.0/0
```

### Unexpected behaviour you saw

On systemd 256 same command results in completly difrrent output and is missing several entries.
```
$systemctl set-property systemd-resolved.service IPAddressDeny= IPAddressAllow= IPAddressDeny=any IPAddressAllow=2.2.2.2 IPAddressAllow=localhost
```
Here several parameters are not present, reset command like `IPAddressDeny=` is not there.
```
# /etc/systemd/system.control/systemd-resolved.service.d/50-IPAddressDeny.conf
# This is a drop-in unit file extension, created via "systemctl set-property"
# or an equivalent operation. Do not edit.
[Service]
IPAddressDeny=0.0.0.0/0
IPAddressDeny=::/0


# /etc/systemd/system.control/systemd-resolved.service.d/50-IPAddressAllow.conf
# This is a drop-in unit file extension, created via "systemctl set-property"
# or an equivalent operation. Do not edit.
[Service]
IPAddressAllow=::1/128
IPAddressAllow=127.0.0.0/8
```

### Steps to reproduce the problem

Use systemctl set-propery with `IPAddressAllow={value}` multiple times or in diffrent time when you want to allow next ip.
```
#start with reset value
$systemctl set-property systemd-resolved.service IPAddressAllow=
#set allowed ips
$systemctl set-property systemd-resolved.service IPAddressAllow=1.1.1.2
$systemctl set-property systemd-resolved.service IPAddressAllow=1.1.1.1
```

After check I end up with.
```
# /etc/systemd/system.control/systemd-resolved.service.d/50-IPAddressAllow.conf
# This is a drop-in unit file extension, created via "systemctl set-property"
# or an equivalent operation. Do not edit.
[Service]
IPAddressAllow=1.1.1.1/32
```
Is this is intended way how is this supposed to be working for properties with multiple entries? _Btw why set-property cannot set all properties? I understand, that some of them needs restart of service/diffrent action, but set-property could just set it and tell user to restart &do diffrent action and allow setting properties directly._

**Comment by yuwata** [comment]
It seems a regression of 84ebe6f01381c21b88e37e856956c9c9ee6781d6. Will fix.

**Comment by yuwata** [comment]
Fix is waiting in #34787.
